jota/src/main/java/org/iota/jota/IotaAPI.java

```diff
@@ -74,186 +77,95 @@ protected IotaAPI(Builder builder) {
-     * @param checksum  Adds 9-tryte address checksum. Checksums are required for all API calls.
-     * @return {@link GetNewAddressResponse}
-     * @throws ArgumentException When the seed is invalid
-     * @throws ArgumentException When the security level is wrong.
-     */
-    public GetNewAddressResponse getNextAvailableAddress(String seed, int security, boolean checksum) throws ArgumentException {
-        return generateNewAddresses(seed, security, checksum, 0, 1, false);
+        return GetNewAddressResponse.create(addresses, stopWatch.getElapsedTimeMili());
     }
-    
-    /**
-     * Checks all addresses until the first unspent address is found.
-     * 
-     * @param seed      Tryte-encoded seed. It should be noted that this seed is not transferred.
-     * @param security  Security level to be used for the private key / address. Can be 1, 2 or 3.
-     * @param checksum  Adds 9-tryte address checksum. Checksums are required for all API calls.
-     * @param index     Key index to start search from.
-     * @return {@link GetNewAddressResponse}
-     * @throws ArgumentException When the seed is invalid
-     * @throws ArgumentException When the security level is wrong.
-     */
-    @Document
-    public GetNewAddressResponse getNextAvailableAddress(String seed, int security, boolean checksum, int index) throws ArgumentException {
-        return generateNewAddresses(seed, security, checksum, index, 1, false);
-    }
-    
-    /**
-     * Generates new addresses, meaning addresses which were not spend from, according to the connected node.
-     * Starts at index 0, until <code>amount</code> of unspent addresses are found.
-     * 
-     * @param seed      Tryte-encoded seed. It should be noted that this seed is not transferred.
-     * @param security  Security level to be used for the private key / address. Can be 1, 2 or 3.
-     * @param checksum  Adds 9-tryte address checksum. Checksums are required for all API calls.
-     * @param amount    Total number of addresses to generate.
-     * @return {@link GetNewAddressResponse}
-     * @throws ArgumentException When the seed is invalid
-     * @throws ArgumentException When the security level is wrong.
-     * @throws ArgumentException When the amount is negative
-     */
-    @Document
-    public GetNewAddressResponse generateNewAddresses(String seed, int security, boolean checksum, int amount) throws ArgumentException {
-        return generateNewAddresses(seed, security, checksum, 0, amount, false);
+
+    private boolean isAddressSpent(String newAddress, boolean checksum) {
+        final String address = checksum ? newAddress : Checksum.addChecksum(newAddress);
+        final FindTransactionResponse response = findTransactionsByAddresses(address);
+
+        if (response.getHashes().length == 0) {
+            WereAddressesSpentFromResponse spent = wereAddressesSpentFrom(address);
+            return spent.getStates().length > 0 && !spent.getStates()[0];
```

Comment:
You can change this to `checkWereAddressSpentFrom` (And maybe rename it to checkWas...)